File: python/demo00.py
'''
INTRODUCTION TO MULTITHREADING
You should try running this app several times and see results.
'''

import threading



def do_task():
    for _ in range(300):
        print('B', end='')



th = threading.Thread(target=do_task)
th.start()

for _ in range(300):
    print('A', end='')

th.join()
print()

File: cpp/cpp-std/demo00.cpp
/*
INTRODUCTION TO MULTITHREADING
You should try running this app several times and see results.
*/


#include <iostream>
#include <thread>
using namespace std;



void doTask() {
    for (int i = 0; i < 300; ++i)
        cout << "B";
}



int main() {
    std::thread th(&doTask);

    for (int i = 0; i < 300; ++i)
        cout << "A";

    th.join();

    cout << endl;
    return 0;
}
